service/models.py

```diff
@@ -1,227 +1,227 @@
-        Deserializes a Shopcart from a dictionary
+        Deserialize a shopcart dictionary to object
 
         Args:
             data (dict): A dictionary containing the resource data
         """
         try:
-            self.customer_id = data["customer_id"]
-            self.customer_name = data["customer_name"]
+            self.id = data["id"]
+            self.name = data["name"]
+            items_js = data["items"]
+            for item_js in items_js:
+                item = Item()
+                item.deserialize(item_js)
+                self.items.append(item)
         except KeyError as error:
             raise DataValidationError(
-                "Invalid Shopcart: missing " + error.args[0]
+                f"Invalid {type(self).__name__}: missing {error.args[0]}"
             ) from error
         except TypeError as error:
             raise DataValidationError(
-                "Invalid Shopcart: body of request contained bad or no data - "
-                "Error message: " + error
+                f"Invalid {type(self).__name__}: failed to deserialize request body\nError message: {error}"
             ) from error
         return self
 
     @classmethod
     def init_db(cls, app):
-        """ Initializes the database session """
-        logger.info("Initializing database")
+        """ Initialize the Shopcart Table """
+        logger.info(f"Start initializing the {cls.__name__} Table")
         cls.app = app
-        # This is where we initialize SQLAlchemy from the Flask app
-        db.init_app(app)
+        db.init_app(app)  # init the Flask app for SQLAlchemy
         app.app_context().push()
         db.create_all()  # make our sqlalchemy tables
+        logger.info(f"Done initializing the {cls.__name__} Table")
 
     @classmethod
-    def all(cls):
-        """ Returns all of the Shopcarts in the database """
-        logger.info("Processing all Shopcarts")
+    def get_all(cls):
+        """ Get all shopcarts in DB table """
+        logger.info(f"Get all {cls.__name__}s")
         return cls.query.all()
 
     @classmethod
-    def find(cls, id):
-        """ Finds a Shopcart by it's ID """
-        logger.info("Processing lookup for id %s ...", id)
+    def get_by_id(cls, id):
+        """ Get shopcart by primary key: id """
+        logger.info(f"Get {cls.__name__} by id={id}")
         return cls.query.get(id)
 
     @classmethod
     def find_by_name(cls, name):
-        """Returns all Shopcarts with the given name
+        """Find shopcart(s) by name
 
         Args:
             name (string): the name of the Shopcarts you want to match
         """
-        logger.info("Processing name query for %s ...", name)
-        return cls.query.filter(cls.customer_name == name) 
-    
+        logger.info(f"Get {cls.__name__} by name={name}")
+        return cls.query.filter(cls.name == name)
+
+    @classmethod
+    def find_id(cls):
+
+        result = cls.query.with_entities(Shopcart.customer_id).all()
```

Comment:
should be `Shopcart.id` instead of `Shopcart.customer_id` now lol (sorry for causing the trouble)

Reply:
Yes, it should be **Shopcart.id** now